lib/file/widgets/file_menu.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:black_hole_flutter/black_hole_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:pedantic/pedantic.dart';
+import 'package:schulcloud/app/app.dart';
+
+import '../data.dart';
+import 'file_thumbnail.dart';
+
+class FileMenu extends StatelessWidget {
+  const FileMenu({this.file});
+
+  final File file;
+
+  Future<void> _delete(BuildContext context) async {
+    unawaited(file.delete());
+    context.rootNavigator.pop();
+  }
+
+  Future<void> _rename(BuildContext context) async {
+    final newName = await showDialog<String>(
+      context: context,
+      useRootNavigator: true,
+      builder: (_) => RenameDialog(oldName: file.name),
+    );
+
+    if (newName == null) {
+      return;
+    }
+
+    unawaited(file.rename(newName));
+    context.navigator.pop();
+  }
```

Comment:
One method uses `context.navigator` and the other one uses `context.rootNavigator`? AFAIK both refer to the same `Navigator` in this case, so probably prefer `context.navigator`?
```suggestion
  Future<void> _delete(BuildContext context) async {
    unawaited(file.delete());
    context.navigator.pop();
  }

  Future<void> _rename(BuildContext context) async {
    final newName = await showDialog<String>(
      context: context,
      useRootNavigator: true,
      builder: (_) => RenameDialog(oldName: file.name),
    );

    if (newName == null) {
      return;
    }

    unawaited(file.rename(newName));
    context.navigator.pop();
  }
```